src/main/java/com/example/onlinebookstore/repository/BookRepository.java

```diff
@@ -1,7 +1,22 @@
 package com.example.onlinebookstore.repository;
 
+import com.example.onlinebookstore.dto.book.BookResponseDto;
 import com.example.onlinebookstore.model.Book;
+import java.util.List;
+import java.util.Optional;
+import org.springframework.data.domain.Pageable;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.stereotype.Repository;
 
+@Repository
 public interface BookRepository extends JpaRepository<Book, Long> {
+    @Query("select b from Book b join fetch b.categories c where c.id = :categoryId")
+    List<Book> findAllByCategoryId(Long categoryId);
+
+    @Query("select b from Book b join fetch b.categories where b.id = :id")
+    Optional<Book> findById(Long id);

```

Comment:
I would explicitly tell that when method fetches some relations. For example `findByIdWithCategories` or similar one. In this case you'll be sure which relations method returns and minimize a possibility to get `LazyInitializationException` or sendind redundant queries to DB instead of one query.